### ISSUE TITLE
`question_mark` doesn't recognize `match` with `None => None` arm

### ISSUE THREAD

**Issue by cakebaker** [rust-lang/rust-clippy#16751]
### Summary

Hi,

I noticed in our code base that a `match` with a `None => None` arm doesn't lead to a warning from the `question_mark` lint.

### Lint Name

question_mark

### Reproducer

I tried this code:

```rust
fn main() {
    let _x = example(Some(12));
}

fn example(v: Option<usize>) -> Option<usize> {
    match v {
        Some(n) => {
            println!("{n}");
            Some(42)
        }
        None => None,
    }
}
```

I expected to see this happen: A warning from the lint when running `cargo clippy -- -W clippy::question_mark
`

Instead, this happened: There was no warning.

### Version

```text
rustc 1.94.0 (4a4ef493e 2026-03-02)
binary: rustc
commit-hash: 4a4ef493e3a1488c6e321570238084b38948f6db
commit-date: 2026-03-02
host: x86_64-unknown-linux-gnu
release: 1.94.0
LLVM version: 21.1.8
```

**Comment by sylvestre** [comment]
@samueltardieu ^^ ;)

**Comment by profetia** [comment]
This should be a FN of `manual_map`

**Comment by xtqqczze** [comment]
what about suggesting:

```rust
fn example(v: Option<usize>) -> Option<usize> {
    let n = v?;
    println!("{n}");
    Some(42)
}
```

**Comment by cakebaker** [comment]
@profetia my example is probably not that great. Here you can see the real code from which I derived the example: https://github.com/uutils/coreutils/pull/11461/changes
